extension/src/openvic-extension/utility/UITools.cpp

```diff
@@ -123,6 +128,69 @@ static Array get_events_from_shortcut_key(String const& key) {
 	return events;
 }
 
+static Error try_create_shortcut_action_for_button(
+	GUIButton* gui_button, String const& shortcut_key_name, String const& shortcut_hotkey_name = ""
+) {
+	Array event_array = get_events_from_shortcut_key(shortcut_key_name);
+
+	ERR_FAIL_COND_V_MSG(
+		!shortcut_key_name.is_empty() && event_array.is_empty(), ERR_INVALID_PARAMETER,
+		vformat("Unknown shortcut key '%s' for GUI button %s", shortcut_key_name, gui_button->get_name())
+	);
+
+	if (shortcut_key_name.is_empty()) {
+		return OK;
+	}
```

Comment:
Not a big deal, but this could go at the very beginning of the function and skip the first two steps.